### PR TITLE
Feature/mobile layout

### PR DESCRIPTION
**Pull request by IvanRodriCalleja**
Style mobile app

### CODE REVIEW

File: client/.babelrc [hunk 1]
```diff
@@ -15,7 +15,11 @@
           "modules": "./app/modules",
           "webSharedComponents": "./app/web/modules/shared",
           "webContent": "./app/webcontent",
-          "webConfig": "./app/web/config.js"
+          "webConfig": "./app/web/config.js",
+          "customFragment": "./app/modules/shared/CustomFragment.js",
+          "mobileApp": "./app/mobile",
+          "mobileSharedComponents": "./app/mobile/modules/shared",
+          "mobileRouter": "./app/mobile/router.js"
```
Comment (franmolmedo) [inline client/.babelrc]: ¿?

File: client/app/mobile/modules/series/root/DetailSerie.js [hunk 2]
```diff
@@ -1,14 +1,42 @@
...
 
 const DetailSerie = ({ serie }) => 
-    <ScrollView>
-        <Info serie={ serie } />
-        <Seassons seassons={ serie.seassons } />
-        <Actors actors={ serie.actors } />
-    </ScrollView>
+    <View style={ styles.container }>
+        <Image style={ styles.backgroundContainer } source={ background } />
+        <View style={ styles.contentContainer }>
```
Comment (franmolmedo) [inline client/app/mobile/modules/series/root/DetailSerie.js]: If we have a customfragment try using it

File: client/app/mobile/modules/series/root/detailSerie/Info.js [hunk 3]
```diff
@@ -1,17 +1,75 @@
 import React from 'react';
-import { View, Image, Text } from 'react-native';
+import { View, 
+    Image, 
+    Text,
+    StyleSheet 
+} from 'react-native';
```
Comment (franmolmedo) [inline client/app/mobile/modules/series/root/detailSerie/Info.js]: Inline declaration when possible

File: client/app/mobile/modules/series/root/detailSerie/Info.js [hunk 4]
```diff
@@ -1,17 +1,75 @@
...
             source={{
                 uri: serie.image.medium,
                 cache: 'only-if-cached',
             }}
         />
-        <Text>{ serie.title }</Text>
-        <Text>{ serie.language }</Text>
+        <View style={ styles.dataContainer }>
+            <Text style={ styles.title }>{ serie.title }</Text>
+            <View style={ styles.dataAlignRow }>
```
Comment (franmolmedo) [inline client/app/mobile/modules/series/root/detailSerie/Info.js]: Are you sure you need all this complex structure?

File: client/app/mobile/modules/series/root/detailSerie/Info.js [hunk 5]
```diff
@@ -1,17 +1,75 @@
...
+    },
+    rightDataChild: {
+        flex: 0.5,
+        paddingLeft: 10
+    },
+    textFormat: {
+        color: '#828080',
+        fontSize: 15
+    }   
+});
```
Comment (franmolmedo) [inline client/app/mobile/modules/series/root/detailSerie/Info.js]: we are not going to talk about styles, but maybe we can order them or use a better naming.

File: client/app/mobile/modules/series/root/detailSerie/Seassons.js [hunk 6]
```diff
@@ -1,10 +1,10 @@
 import React from 'react';
-import { View } from 'react-native';
+import { View, StyleSheet } from 'react-native';
 import SeassonSF from 'modules/series/root/detailSerieFS/SeassonSF';
 import Seasson from './seassons/Seasson';
 
 const Seassons = ({ seassons }) => 
-    <View>
+    <View style={ styles.container }>
```
Comment (franmolmedo) [inline client/app/mobile/modules/series/root/detailSerie/Seassons.js]: Idem as before. If we can render the child, try not generating a complex view structure

File: client/app/mobile/modules/series/root/detailSerie/actors/Actor.js [hunk 7]
```diff
@@ -1,26 +1,63 @@
...
+                />
+                <Text style={ styles.actorNameText }>{ person.name }</Text>
+            </View>
+            <View>
+                <Image 
+                    source={ exchangeIcon }
+                    style={ styles.exchangeImage }
+                />
+            </View>
+            <View style={ styles.subContainer }>
```
Comment (franmolmedo) [inline client/app/mobile/modules/series/root/detailSerie/actors/Actor.js]: Container... subcontainer... use names that remind the concept.

File: client/app/mobile/modules/series/root/masterSeries/seriesGrid/SerieCard.js [hunk 8]
```diff
@@ -2,25 +2,77 @@ import React from 'react';
...
-                <Text>{ serie.title }</Text>
+                <View style={ styles.dataContainer }>
+                    <Text style={ styles.title }>{ serie.title }</Text>
+                    <View style={ styles.dataAlignRow }>
+                        <View style={ styles.leftDataChild }>
+                            <Text style={ styles.textFormat }>{ serie.language }</Text>
+                        </View>
+                        <View style={ styles.rightDataChild }>
+                            <Text style={ styles.textFormat }>Episodios: 100</Text>
+                            <Text style={ styles.textFormat }>Temporadas: 6</Text>
```
Comment (franmolmedo) [inline client/app/mobile/modules/series/root/masterSeries/seriesGrid/SerieCard.js]: Very complex. Why using numbers? I think for a pr is better setting the values in variables and then only change them for props variables.

File: client/app/modules/series/root/MasterSeriesSF.js [hunk 9]
```diff
@@ -1,8 +1,8 @@
-import React, { Component, Fragment } from 'react';
+import React, { Component } from 'react';
 import { connect } from 'react-redux';
 import { getAllSeriesRequest } from 'state/series/actions/getSeries';
 import { searchSeriesRequest } from 'state/series/actions/searchSeries';
-
+import CustomFragment from 'customFragment';
```
Comment (franmolmedo) [inline client/app/modules/series/root/MasterSeriesSF.js]: Let´s import utils right behind libraries

File: client/app/state/series/sagas/getSerie.js [hunk 10]
```diff
@@ -1,10 +1,15 @@
...
 
-function *getSerieAction(action) {
+
+function *getSerieAction(/*action*/) {
     try {
-        const serie= yield call(getSerieById, action.payload);
+        //const serie= yield call(getSerieById, action.payload);
+        const serie = {
+            data: fakeSeries
+        }
```
Comment (franmolmedo) [inline client/app/state/series/sagas/getSerie.js]: Remove all comments

File: client/app/state/series/sagas/getSeries.js [hunk 11]
```diff
@@ -1,10 +1,15 @@
...
 function *getAllSeriesAction() {
     try {
-        const series= yield call(getSeries);     
+        //const series= yield call(getSeries);   
+        const series = {
+            data: fakeSeries
+        }  
         yield put(getAllSeriesDone(series))
     } catch (error) {
         yield put(getAllSeriesError(error))
```
Comment (franmolmedo) [inline client/app/state/series/sagas/getSeries.js]: Remove comments

File: client/app/state/series/sagas/searchSeries.js [hunk 12]
```diff
@@ -1,10 +1,14 @@
...
+import fakeSeries from '../../../../search.json';
 
-function *seachSeriesAction(action) {
+function *seachSeriesAction(/*action*/) {
     try {
-        const series = yield call(searchSeriesByTitle, action.payload);     
+        //const series = yield call(searchSeriesByTitle, action.payload);     
+        const series = {
+            data: fakeSeries
+        }
```
Comment (franmolmedo) [inline client/app/state/series/sagas/searchSeries.js]: Remove

File: client/app/mobile/modules/series/root/masterSeries/seriesGrid/SerieCard.js [hunk 13]
```diff
@@ -2,25 +2,77 @@ import React from 'react';
...
+                    </View>
+                </View>
             </View>        
         </Link>      
     )     
 }
 
-export default SerieCard;
\ No newline at end of file
+export default SerieCard;
```
Comment (franmolmedo) [inline client/app/mobile/modules/series/root/masterSeries/seriesGrid/SerieCard.js]: Why default exporting? It is better using named export

File: client/app/mobile/modules/series/root/masterSeries/SeriesGrid.js [hunk 14]
```diff
@@ -12,12 +13,19 @@ class SeriesGrid extends Component {
     render() {
         return(
             <FlatList 
-                data={this.props.series}
+                style={ styles.container }
+                data={ this.props.series }
                 renderItem={ ({ item }) => this.renderSerieCard(item) }
-                keyExtractor={(item) => item.id}
+                keyExtractor={ (item) => item.id }
```
Comment (franmolmedo) [inline client/app/mobile/modules/series/root/masterSeries/SeriesGrid.js]: Be consistent if you are going to add spaces or not. There are other places where you don´t use blanks

File: client/app/mobile/modules/series/root/detailSerie/seassons/seasson/Episode.js [hunk 15]
```diff
@@ -1,16 +1,37 @@
 import React from 'react';
-import { View, Text, Image } from 'react-native';
+import { View, Text, Image, StyleSheet } from 'react-native';
 
 const Episode = ({ episode }) => 
-    <View>
+    <View style={ styles.container }>
         <Image 
-            style={{ width: 150, height: 150 }}
+            style={ styles.image }
```
Comment (franmolmedo) [inline client/app/mobile/modules/series/root/detailSerie/seassons/seasson/Episode.js]: Do you need that? Use proper CSS

File: client/app/mobile/modules/series/root/detailSerie/seassons/Seasson.js [hunk 16]
```diff
@@ -1,24 +1,65 @@
...
+    Text, 
+    TouchableWithoutFeedback, 
+    StyleSheet, 
+    Image 
+} from 'react-native';
 import Episode from './seasson/Episode';
+import upIcon from 'content/icons/up-arrow.png';
+import downIcon from 'content/icons/down-arrow.png';
 
 const Seasson = function() {
```
Comment (franmolmedo) [inline client/app/mobile/modules/series/root/detailSerie/seassons/Seasson.js]: FAF

File: client/app/mobile/modules/series/root/detailSerie/seassons/Seasson.js [hunk 17]
```diff
@@ -1,24 +1,65 @@
...
+                    </View>
+                </View>
+                <View>
+                    { this.state.isOpen ? this.props.seasson.episodes.map(episode => 
+                        <Episode 
+                            key={ episode.title } 
+                            episode={ episode } 
+                        />) : null
+                    }
+                </View>
```
Comment (franmolmedo) [inline client/app/mobile/modules/series/root/detailSerie/seassons/Seasson.js]: Destructure props

File: client/app/mobile/modules/series/root/detailSerie/seassons/Seasson.js [hunk 18]
```diff
@@ -1,24 +1,65 @@
...
+                        <View>
+                            <Image 
+                                source={ this.state.isOpen ? upIcon : downIcon }
+                                style={ styles.image }
+                            />
+                        </View>
+                    </View>
+                </View>
+                <View>
+                    { this.state.isOpen ? this.props.seasson.episodes.map(episode =>
```
Comment (franmolmedo) [inline client/app/mobile/modules/series/root/detailSerie/seassons/Seasson.js]: this.state? func comp?

File: client/app/mobile/modules/series/root/detailSerie/seassons/Seasson.js [hunk 19]
```diff
@@ -1,24 +1,65 @@
...
+            <View style={ styles.container }>
+                <View style={ styles.seassonRow }>
+                    <Text style={ styles.seassonText }>Seasson { this.props.seasson.seasson }</Text>
+                    <View>
+                        <View>
+                            <Image 
+                                source={ this.state.isOpen ? upIcon : downIcon }
+                                style={ styles.image }
+                            />
+                        </View>
```
Comment (franmolmedo) [inline client/app/mobile/modules/series/root/detailSerie/seassons/Seasson.js]: Image inside a view, inside a view?

File: client/app/mobile/modules/series/root/DetailSerie.js [hunk 20]
```diff
@@ -1,14 +1,42 @@
...
+                <Info serie={ serie } />
+                <Seassons seassons={ serie.seassons } />
+                <Actors actors={ serie.actors } />
+            </ScrollView>
+        </View>
+    </View>
 
-export default DetailSerie;
\ No newline at end of file
+export default DetailSerie;
```
Comment (franmolmedo) [inline client/app/mobile/modules/series/root/DetailSerie.js]: Avoid default exports when possible

**Review by franmolmedo — CHANGES_REQUESTED** [review]

File: client/app/mobile/modules/series/root/detailSerie/Info.js (hunk 5 above)
Comment (franmolmedo) [reply client/app/mobile/modules/series/root/detailSerie/Info.js]: Also, you can use extended stylesheet or something to share common values.

File: client/app/mobile/modules/series/root/masterSeries/seriesGrid/SerieCard.js (hunk 8 above)
Comment (IvanRodriCalleja) [reply client/app/mobile/modules/series/root/masterSeries/seriesGrid/SerieCard.js]: I am waiting to have the API to use real data, right now i have not this information in my fakeData

File: client/app/mobile/modules/series/root/detailSerie/Info.js (hunk 4 above)
Comment (IvanRodriCalleja) [reply client/app/mobile/modules/series/root/detailSerie/Info.js]: This complex structure is for layout

File: client/app/state/series/sagas/getSerie.js (hunk 10 above)
Comment (IvanRodriCalleja) [reply client/app/state/series/sagas/getSerie.js]: I will remove the coments when the API exist, at the moment is the way i am using in mobile development, i can make a localhost request but it only works in ipv4 and i have to change continuously when y chane my network

File: client/app/state/series/sagas/getSeries.js (hunk 11 above)
Comment (IvanRodriCalleja) [reply client/app/state/series/sagas/getSeries.js]: Same as before

File: client/app/state/series/sagas/searchSeries.js (hunk 12 above)
Comment (IvanRodriCalleja) [reply client/app/state/series/sagas/searchSeries.js]: Same as before

**Pull request merged**